vitest.config.ts

```diff
@@ -8,5 +8,21 @@ export default defineConfig({
     // you might want to disable it, if you don't have tests that rely on CSS
     // since parsing CSS is slow
     css: true,
+    environmentOptions: {
+      jsdom: {
+        /*
+          We need to set the url parameter below to the url of the API server 
+          to avoid CORS issue.
+          
+          This is because JSDom environment sets the 
+          origin header (axios cannot control it on its own, just like in a 
+          browser) automatically to the url specified here. 
+          
+          If this url is not provided then vitest sets the default value as 
+          localhost:3000 which causes CORS error with the test server.
+        */
+        url: 'http://localhost:55001',
```

Comment:
CORS error appears in tests when superagent is replaced with axios in `resetFixtures.ts`. Not sure why that would be, but I inspcted the API requests being made with fiddler and requests made with superagent did not contain the `Origin` header (used to enforce CORS) while requests made with axios did contain this header.

Setting the window URL to `localhost:55001` for JSDom environment fixes the issue.

Reply:
Ok, I understand now the CORS error happened in `resetFixtures.ts`.